code/modules/antagonists/changeling/changeling.dm

```diff
@@ -392,17 +392,31 @@
 		to_chat(owner.current, span_warning("We lack the energy to evolve new abilities right now!"))
 		return FALSE
 
-	var/datum/action/changeling/new_action = new sting_path()
+	var/success = give_power(sting_path)
+	if(success)
+		genetic_points -= initial(sting_path.dna_cost)
+	return success
```

Comment:
The `give_power()` proc that you introduce in this PR doesn't return TRUE at all, so this will _always_ be null, making line 397 unreachable. It will also mean that we always return a null value (falsey), which isn't nice to any consumers of this proc.

On second read, you document that give_power returns TRUE, but you appear to have forgotten to actually code that in. Please do fix this.